### PR TITLE
Session fixes

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: controller/handler_edge_ctrl/common_tunnel.go [hunk 1]
```diff
@@ -180,7 +180,9 @@ func (self *baseTunnelRequestContext) ensureSessionForService(sessionId, session
 		self.session, err = self.handler.getAppEnv().Handlers.Session.Read(id)
 		if err != nil {
 			err = internalError(err)
+			return
 		}
+		self.newSession = true
```
Comment (andrewpmartinez) [inline controller/handler_edge_ctrl/common_tunnel.go:185]: `newApiSession` ?

Comment (plorenz) [reply controller/handler_edge_ctrl/common_tunnel.go:185]: It's marking that a new session was created in ensureSessionForService. ApiSessions are created in ensureApiSession

Comment (andrewpmartinez) [reply controller/handler_edge_ctrl/common_tunnel.go:185]: I didn't unfold the code enough.

Comment (plorenz) [reply controller/handler_edge_ctrl/common_tunnel.go:185]: yeah, with the default context it looks api session related

**Review by andrewpmartinez — APPROVED** [review]
Seems like a fine near-term fix, but if we get the chance to formalize the error cleanup process, that might be nice.

**Pull request merged**